VOLTTRON/volttron
# Feature/recipes mergeable

# Description

Adds the initial components for the deployment recipes system. It establishes the deployment directory with a standard ansible organization structure, which includes a number of available roles and three playbooks which build on those roles. This system is intended to facilitate the management of remote systems, especially when that involves multiple remote platforms. The components provided constitute fully useful playbooks, but also provides building blocks for users who may want to construct more specialized playbooks.

## Type of change

Please delete options that are not relevant.


# How Has This Been Tested?

Because this system is inherently about working with remote systems, it is not clear how to test other than as an integration test. I've been working on a research cloud with multiple VMs, from my controller system I configure 5 remote platforms, each on its own VM:

* collector 1: RMQ on Ubuntu 18.04
* collector 2: ZMQ on Ubuntu 18.04
* collector 3: RMQ on CentOS 7
* collector 4: ZMQ on CentOS 7
* web: ZMQ on Ubuntu 18.04 - with web feature enabled, uses the controller's local volttron source tree

After newly provisioning the VMs, I copy my local ssh key to each (ssh-copy-id) and then run the three playbooks:
1. ensure-host-keys (optional, updates local known-hosts file for the VMs, this probably is done when you copy your ssh public keys, but is done as a test; it produces changes because it is more complete than what ssh-copy-id triggers)
2. host-config
3. install-platform

After running the above and seeing no errors reported, I ssh to each host & activate the volttron venv. I then start the platform and observe that I can use `vctl` to interact with it. There are currently no agents installed (hopefully coming soon).

# Checklist:


# Notes
This does *not* constitute the deployment recipes being fully feature-complete, but it is a foundation that should be usable as is, and which additional features can be added on top of. If we don't want to merge it into develop until the feature set is more complete, I would still like a review of this portion while the size is tractable. I'm fine with moving this PR onto a staging branch if that's desired.
